otri/database/postgresql_adapter.py

```diff
@@ -4,160 +4,101 @@
-            self.__create_table(data.category)
-
-        if(type(data.values) == list):
-            data_json_list = [(json.dumps(element),)
-                              for element in data.values]
-            execute_values(self.cursor, "INSERT INTO {} (data_json) VALUES %s".format(
-                data.category), data_json_list)
-            self.connection.commit()
-            log.v("Upload completed")
-        elif(type(data.values) == dict):
-            self.cursor.execute("INSERT INTO {} (data_json) VALUES %s".format(
-                data.category), (data.values,))
-            self.connection.commit()
-            log.v("Upload completed")
-        else:
-            raise ValueError("Data value not a list or a dict")
-
-    def read(self, query: DatabaseQuery):
+    def __init__(self, password: str, host: str = "localhost", port: Union[str, int] = 5432,
+                 user: str = "postgres", database: str = "postgres"):
         '''
-        Queries the database for the requested values.
-        TODO: Define how a postgre JSON query should be formatted (https://devhints.io/postgresql-json)
-
         Parameters:
-            query : DatabaseQuery
-                Executes the query on the given query.category and the given query.filters
-        Returns:
-            list containing json dicts
+            host : str\n
+                The database host address. Defaults to "localhost".\n
+            port : Union[str, int]\n
+                The port on which the database is listening. Defaults to 5432.\n
+            user : str\n
+                The username. Defaults to "postgres".\n
+            password : str\n
+                The password for the given user. Has no default value.\n
+            database : str\n
+                The database on which to connect. Defaults to "postgres".
         '''
-        self.cursor.execute("SELECT data_json as json FROM {} WHERE {};".format(
-            query.category, query.filters))
-        return [json.dumps(element[0]) for element in self.cursor.fetchall()]
+        super().__init__(password, host, port, user, database)
 
-    def stream(self, query: DatabaseQuery, batch_size: int = 1000) -> PostgreSQLStream:
+    def stream(self, query, batch_size: int = 1000) -> PostgreSQLStream:
         '''
-        Returns a database stream.
+        Returns a database stream that performs the given query fetching `batch_size` rows at a
+        time. If you need to fetch few rows at a time but do not need a `Stream` object use
+        sqlalchemy's `yield_per(amount)` in a session.
+        A new connection is opened for each stream.
 
         Parameters:
-            query : DatabaseQuery
-                Executes the query on the given query.category and the given query.filters
-            batch_size : int
-                The number of rows the database should load before making them available.
-                The iterable still always yields a single item.
+            query\n
```

Comment:
I guess you avoided type-ing the `query` parameter because there's no way to say that it has to be a sqlalchemy object right?

Reply:
As a matter of fact I remembered that the Query type is a thing so yes I could, I'll change them

Reply:
Only the table has no real unique type from what I understand